### DISCUSSION TITLE
Allow other using alias to be used to the right of the equal sign of the using alias.

### DISCUSSION THREAD

**Discussion by Summpot** [dotnet/csharplang#7681] [Language Ideas] [answered]
The following writeup is not supported in c# now.
``` csharp
using A = int;
using unsafe B = delegate* unmanaged<A>;
```

It will report an error.
`error CS0246: The type or namespace name 'A' could not be found (are you missing a using directive or an assembly reference?)`

As in the example I gave, this also enables named function pointers.

Since using alias will be expanded at compile time, the following problem mentioned in the proposal for function pointers seems to no longer exist.
> Part of the problem here is the underlying CLI primitive doesn't have names hence this would be purely a C# invention and require a bit of metadata work to enable. That is doable but is a significant about of work. It essentially requires C# to have a companion to the type def table purely for these names.

**Comment by ufcpp** [comment, marked as answer]
https://github.com/dotnet/csharplang/discussions/3873
